src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -0,0 +1,675 @@
+  }
+
+  if( FD_UNLIKELY( in_idx==STORE_IN_IDX ) ) {
+    if( FD_UNLIKELY( chunk<ctx->repair_req_in_chunk0 || chunk>ctx->repair_req_in_wmark ) )
+      FD_LOG_ERR(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz,
+            ctx->repair_req_in_chunk0, ctx->repair_req_in_wmark ));
+
+    uchar const * dcache_entry = fd_chunk_to_laddr_const( ctx->repair_req_in_mem, chunk );
+    handle_new_repair_requests( ctx, dcache_entry, sz );
+    return;
+  }
+  
+  if( FD_UNLIKELY( chunk<ctx->chunk || chunk>ctx->wmark || sz>FD_NET_MTU ) ) {
+    FD_LOG_ERR(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz, ctx->chunk, ctx->wmark ));
+    *opt_filter = 1;
+    return;
+  }
+
+  uchar const * dcache_entry = fd_chunk_to_laddr_const( ctx->net_in, chunk );
+  ulong  hdr_sz = fd_disco_netmux_sig_hdr_sz( sig );
+  FD_TEST( hdr_sz < sz ); /* Should be ensured by the net tile */
+
+  ctx->repair_buffer_sz = sz;
+  fd_memcpy( ctx->repair_buffer, dcache_entry, sz );
+
+  *opt_filter = 0;
+
+  return;
+}
+
+static void
+after_frag( void *             _ctx,
+            ulong              in_idx,
+            ulong              seq        FD_PARAM_UNUSED,
+            ulong *            opt_sig,
+            ulong *            opt_chunk  FD_PARAM_UNUSED,
+            ulong *            opt_sz     FD_PARAM_UNUSED,
+            ulong *            opt_tsorig FD_PARAM_UNUSED,
+            int *              opt_filter,
+            fd_mux_context_t * mux        FD_PARAM_UNUSED) {
+
+  fd_repair_tile_ctx_t * ctx = (fd_repair_tile_ctx_t *)_ctx;
+
+  *opt_filter = 1;
+
+  if( FD_UNLIKELY( in_idx==CONTACT_IN_IDX ) ) {
+    return;
+  }
+
+  if( FD_UNLIKELY( in_idx==STAKE_IN_IDX ) ) {
+    return;
+  }
+
+  if( FD_UNLIKELY( in_idx==STORE_IN_IDX ) ) {
+    return;
+  }
+
+  uint   ip = fd_disco_netmux_sig_dst_ip( *opt_sig );
+  ulong hdr_sz = fd_disco_netmux_sig_hdr_sz( *opt_sig );
+  eth_ip_udp_t * hdr = (eth_ip_udp_t *)ctx->repair_buffer;
+
+  fd_repair_peer_addr_t peer_addr;
+  peer_addr.l = 0;
+  peer_addr.addr = ip;
+  peer_addr.port = fd_ushort_bswap( hdr->udp->net_sport );
+
+  fd_repair_settime( ctx->repair, fd_log_wallclock() );
+  fd_repair_continue( ctx->repair );
+  fd_repair_recv_packet( ctx->repair, ctx->repair_buffer + hdr_sz, ctx->repair_buffer_sz - hdr_sz, &peer_addr );
+}
+
+static void
+privileged_init( fd_topo_t *      topo    FD_PARAM_UNUSED,
+                 fd_topo_tile_t * tile    FD_PARAM_UNUSED,
+                 void *           scratch FD_PARAM_UNUSED) {
+
+}
+
+static void
+during_housekeeping( void * _ctx ) {
+  fd_repair_tile_ctx_t * ctx = (fd_repair_tile_ctx_t *)_ctx;
+
+  fd_mcache_seq_update( ctx->store_out_sync, ctx->store_out_seq );
+
+  fd_repair_settime( ctx->repair, fd_log_wallclock() );
+  fd_repair_continue( ctx->repair );
+}
+
+static void
+unprivileged_init( fd_topo_t *      topo,
+                   fd_topo_tile_t * tile,
+                   void *           scratch ) {
+  fd_flamenco_boot( NULL, NULL );
```

Comment:
I wouldn't trust this glibc crap to succeed in the sandbox, does this need it? Much better to not use this.